### ISSUE TITLE
automatic network type circulation?

### ISSUE THREAD

**Issue by takiainen** [zamojski/TowerCollector#27]
I don't if it is possible, but what if the app could rotate the network type automatically i.e. 2G --> 3G --> 4G and back to 2G with some preselected interval and thus make the collection more efficient. Then users would not need to go the same routes over and over again after manually selecting different network type.

**Comment by ehem** [comment]
I doubt this can be done without root access.  Most likely this will also cause a (potentially brief) disruption in cell service and be troublesome to _most_ casual users.  This could be valuable to hard-core cell mappers though.  I must _strongly_ advise this defaulting to off if implemented.

**Comment by takiainen** [comment]
I agree that the default should be off, but I would not be surprised if majority of new data actually comes from "hard core" mappers i.e. following the Pareto principle like so many other things.

**Comment by zamojski** [comment]
That would be a great feature! However it's impossible to be done without root as @ehem mentioned. Regular Android disallows any changes to this section due to security reasons. Anyway its possible on rooted devices as described on [Stack Overflow](http://stackoverflow.com/a/38926356/1929088). I'm not sure if this will be really beneficial as it takes some time to switch between cell towers of different technologies.

**Comment by Lee-Carre** [comment]
> it takes some time to switch between cell[s] of different [protocol]s.

That, and for capable devices capturing neighbouring cells is valuable.

While auto-cycling would be lovely for dedicated surveying sessions, the most reliable way is to re-traverse the same route once for each mode (thrice, else 4 times for 5G).

A technique / strategy I use, is to set a different mode for the return journey, than for the outbound one.

Else, when following the same route across multiple days (e.g. ones daily commute, or other frequented destinations), set different modes on different days.

Compare with how one makes a point of taking a variety of different routes between destinations, to increase the area being sampled.

So, while not automated, it does cycle through the various modes and capture comprehensive data _eventually_, without having to do a lot of excess driving / walking.

**Comment by zamojski** [comment]
Closing as not planned, requires hacking the functionality of Android which may vary from version to version and requires rooting the device.
